src/components/CommonComponents/Pagination/utils/buildComponentData.ts

```diff
@@ -0,0 +1,92 @@
+import { createElement, MouseEvent, ReactElement } from 'react';
+import { LocalizedLink as Link } from 'gatsby-theme-i18n';
+import { FormattedMessage } from 'react-intl';
+import { PageType } from './buildPaginationModel';
+
+type OnClick = {
+  (event: MouseEvent): void;
+};
+
+interface Props extends Partial<HTMLElement> {
+  as: string;
+  onClick?: OnClick;
+  to?: string;
+}
+
+interface ComponentData {
+  props: Props;
+  key: string;
+  content: string | ReactElement;
+}
+
+export function buildComponentData(
+  page: PageType,
+  hrefBuilder: (n: number) => string,
+  onClick: OnClick
+): ComponentData {
+  const props: Props = { as: '', className: page?.className ?? '' };
+  let content: string | ReactElement = '';
+  let key = '';
+
+  // eslint-disable-next-line default-case
```

Comment:
Please add a default case 🙏 (It could even just have a `break`) but it must have one.

Reply:
fixed